code/modules/bayinstruments/event_manager.dm

```diff
@@ -0,0 +1,88 @@
+/datum/musical_event
+	var/sound/object
+	var/mob/subject
+	var/datum/sound_player/source
+	var/time = 0
+	var/new_volume = 100
+
+
+/datum/musical_event/New(datum/sound_player/source, mob/subject, sound/object, time, volume)
+	src.source = source
+	src.subject = subject
+	src.object = object
+	src.time = time
+	src.new_volume = volume
+
+
+/datum/musical_event/proc/tick()
+	if (!(istype(object) && istype(subject) && istype(source))) 
+		return
+	if (src.new_volume > 0) src.update_sound()
+	else src.destroy_sound()
+
+
+/datum/musical_event/proc/update_sound()
+	src.object.volume = src.new_volume
+	src.object.status |= SOUND_UPDATE
+	if (src.subject)
+		src.subject << src.object
+
+
+/datum/musical_event/proc/destroy_sound()
+	if (src.subject)
+		var/sound/null_sound = sound(channel=src.object.channel, wait=0)
+		if (global.musical_config.env_settings_available)
+			null_sound.environment = -1
+		src.subject << null_sound
+	if (src.source || src.source.song)
+		src.source.song.free_channel(src.object.channel)
+
+
+/datum/musical_event_manager
+	var/list/datum/musical_event/events = list()
+	var/suspended = 0
+	var/active = 0
+	var/kill_loop = 0
+
+
+/datum/musical_event_manager/proc/push_event(datum/sound_player/source, mob/subject, sound/object, time, volume)
+	if (istype(source) && istype(subject) && istype(subject) && istype(object) && volume >= 0 && volume <= 100)
+		src.events += new /datum/musical_event(source, subject, object, time, volume)
+
+
+/datum/musical_event_manager/proc/activate()
+	if (active)	return 0
```

Comment:
Another case of valid code but bad style - best to create a new line and tab for the return to conform with most of our code.